### PR TITLE
doc: change the structure of doc files

### PR DESCRIPTION
**Pull request by b-gent**
Reorganizing the doc/nrf folder to make it easier to locate specific RST files whenever they need modification.

The folder now has a clear structure of subfolders rather than having most of the files on the main level.

Scope of work:

- Moved files for different user guides from the main level to respective subfolders (protocols, working with, app_dev).
  If a user guide has chapters made of more than one RST file, additional subfolders are created.
- This change means we no longer need all the vague prefixes in the file names - 'ug', 'gs' etc.
  File names have been shortened and simplified.
- Renamed the parent file of these user guides to index.rst for easier identification.
- Created an 'images' folder for every such folder.
  Images now reside next to the RST files in which they are used rather than in a common folder.
- Added a new top-level section for Ecosystem integration guides, with its respective folder.
- Edited many RST files to fix broken paths to images after they have been moved.
- Fixed broken includes leading to files that have been relocated.
- Removed all Visio source files (.vsdx) for the svg images. These binary files will now be stored outside of Github.

Signed-off-by: Bartosz Gentkowski <bartosz.gentkowski@nordicsemi.no>

### CODE REVIEW

**Review by gmarull — CHANGES_REQUESTED** [review]
I think this still needs another round of simplification. For example, all `ug_`, `app_` prefixes should be deleted. Some other names shortened (remember they end up being URLs), e.g. `protocols/ble/bt_mesh/ug_bt_mesh.rst` -> `protocols/ble/mesh/index.rst`, `protocols/ble/bt_mesh/ug_bt_mesh_concepts.rst` -> `protocols/ble/mesh/concepts.rst`

**Comment by umapraseeda** [comment]
@b-gent, Added the commit for the wrapper page.

**Review by tomchy — APPROVED** [review]

**Review by erikrobstad — CHANGES_REQUESTED** [review]
The Audio DK images should not be there.
Maybe a rebase gone wrong?

**Review by greg-fer — CHANGES_REQUESTED** [review]
Two things:
- The reintroduced "nrf5340_audio_dk" images should be removed. These were removed in a recent PR to sdk-nrf, so most likely your rebase didn't catch them.
- I know @gmarull mentioned shortening the names of some files, but this led to file names like "dcl.rst" or "dfu.rst" for Matter. DCL (Distributed Compliance Ledger) is a CSA database that could be potentially used by other protocols, while DFU is a more broad concept. Thus, there might be more pages named like this in other parts of the doc. Are we ok with that?

**Comment by b-gent** [comment]
> Two things:
> 
> * The reintroduced "nrf5340_audio_dk" images should be removed. These were removed in a recent PR to sdk-nrf, so most likely your rebase didn't catch them.
> * I know @gmarull mentioned shortening the names of some files, but this led to file names like "dcl.rst" or "dfu.rst" for Matter. DCL (Distributed Compliance Ledger) is a CSA database that could be potentially used by other protocols, while DFU is a more broad concept. Thus, there might be more pages named like this in other parts of the doc. Are we ok with that?

The idea is that file path is part of the file name. So yes, there are now a few files called dfu.rst but you can tell by their path what protocol/user guide they belong to. It's a situation similar to our sample files - they are all called README.rst and that's ok because they are identified by their location in the folder structure.

File: doc/nrf/protocols/bt_mesh/vendor_model/vendor_model_chat_sample_walk_through.rst [hunk 1]
```diff
@@ -8,14 +8,14 @@ The sample implements a vendor model that is called the :ref:`bt_mesh_chat_clien
 
 The code below shows how Company ID, Model ID and message opcodes for the :ref:`messages <bt_mesh_chat_client_model_messages>` are defined in the Chat Client model:
 
-.. literalinclude:: ../../samples/bluetooth/mesh/chat/include/chat_cli.h
```
[the remark this answers is not in the document]
Comment (b-gent) [reply doc/nrf/protocols/bt_mesh/vendor_model/vendor_model_chat_sample_walk_through.rst:11]: no, will fix

File: doc/nrf/protocols/bt_mesh/vendor_model/vendor_model_dev_overview.rst [hunk 2]
```diff
@@ -289,7 +289,7 @@ Defining model callbacks
 The Access API provides a set of callbacks that are called when certain events occur.
 These callbacks are defined in :c:struct:`bt_mesh_model_cb`.
 
-.. figure:: /images/bt_mesh_model_callbacks.svg
```
[the remark this answers is not in the document]
Comment (b-gent) [reply doc/nrf/protocols/bt_mesh/vendor_model/vendor_model_dev_overview.rst:292]: no, will fix

**Comment by b-gent** [comment]
> The Audio DK images should not be there. Maybe a rebase gone wrong?

These are now removed.

File: doc/nrf/ecosystem_integration.rst [hunk 3]
```diff
@@ -0,0 +1,23 @@
+.. _ecosystem_integration:
+
+Ecosystems integration
+######################
+
+The following user guides describe the integration of various eco systems and tools with |NCS|.
+The |NCS| is also integrated with the following ecosystems that are documented in private repositories:
+
+* Apple Homekit
+* Apple FindMy
```
Comment (peknis) [inline doc/nrf/ecosystem_integration.rst:10]: A
```suggestion
* Apple Find My
```

File: doc/nrf/ecosystem_integration.rst [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+* ANT
+
+In the case of HomeKit and FindMy, MFi licensees can get access to the repository by issuing a Nordic `DevZone`_ private ticket.
```
Comment (peknis) [inline doc/nrf/ecosystem_integration.rst:13]:
```suggestion
In the case of HomeKit and FindMy, MFi licensees can get access to the repository by issuing a Nordic `DevZone`_ private ticket.
```
```suggestion
In the case of HomeKit and Find My, MFi licensees can get access to the repository by issuing a Nordic `DevZone`_ private ticket.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Comment by umapraseeda** [comment]
Now since https://github.com/nrfconnect/sdk-nrf/pull/10182 is merged, we might need another PR in nrfxlib to remove the Visio source files there.

**Review by gautesl — APPROVED** [review]

**Comment by umapraseeda** [comment]
For [doc/nrf/app_power_opt.rst → doc/nrf/app_dev/optimizing/power_opt.rst](https://github.com/nrfconnect/sdk-nrf/pull/9561/files#diff-477f63e7ee0615ac9d66ca84b7213eb2753acaa1825aaf3c888e54bc6227defd) , do we need the _opt suffix for the new file name?

**Review by gmarull — APPROVED** [review]
overall lgtm, tested a few redirects and they work as expected

**Comment by umapraseeda** [comment]
doc/nrf/images/ncs-west-repos.png → doc/nrf/developing/images/ncs-west-repos.png. I don't think this PNG image is being used. There is a svg equivalent that is used in the docs.

File: doc/nrf/ecosystem_integration.rst [hunk 5]
```diff
@@ -0,0 +1,23 @@
+.. _ecosystem_integration:
+
+Ecosystems integration
+######################
+
+The following user guides describe the integration of various eco systems and tools with |NCS|.
```
Comment (umapraseeda) [inline doc/nrf/ecosystem_integration.rst:6]:
```suggestion
The following user guides describe the integration of various ecosystems and tools with |NCS|.
```

File: doc/_utils/redirects.py [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+    ("dm_ncs_distro", "developing/ncs_distro"),
+    ("doc_build", "documentation/build"),
+    ("doc_build_process", "documentation/build_process"),
+    ("doc_structure", "documentation/structure"),
+    ("doc_styleguide", "documentation/styleguide"),
+    ("doc_templates", "documentation/templates"),
+    ("ug_bt_coex", "external_comp/bt_coex"),
+    ("ug_bt_fast_pair", "external_comp/bt_fast_pair"),
+    ("ug_edge_impulse", "external_comp/edge_impulse"),
+	("ug_memfault", "external_comp/memfault"),
```
Comment (umapraseeda) [inline doc/_utils/redirects.py:52]:
```suggestion
    ("ug_edge_impulse", "external_comp/edge_impulse"),
    ("ug_memfault", "external_comp/memfault"),
```

**Comment by umapraseeda** [comment]
[ /images/matter_components_DM_doorlock.svg → .../images/matter_components_DM_doorlock.svg ](https://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-9561/nrf/protocols/matter/overview/data_model.html#data-model-example-door-lock) is missing the caption.

**Comment by umapraseeda** [comment]
Under protocols\thread\overview\images\, ot-arch_2x.svg (the PNG version is used), thread_platform_design_nRF53.odg and thread_platform_design_nRF53_multi.odg are not used anywhere. May be we can remove these.

File: doc/_utils/redirects.py [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+    ("ug_thread_overview","protocols/thread/overview/index"),
+    ("ug_thread_ot_integration","protocols/thread/overview/ot_integration"),
+    ("ug_thread_ot_memory","protocols/thread/overview/ot_memory"),
+    ("ug_thread_supported_features","protocols/thread/overview/supported_features"),
+    ("ug_thread_prebuilt_libs","protocols/thread/prebuilt_libs"),
+    ("ug_thread_tools","protocols/thread/tools"),
+    ("ug_wifi","protocols/wifi/index"),
+    ("ug_zigbee_adding_clusters","protocols/zigbee/adding_clusters"),
+    ("ug_zigbee_architectures","protocols/zigbee/architectures"),
+    ("ug_zigbee_commissioning","protocols/zigbee/commisioning"),
```
Comment (umapraseeda) [inline doc/_utils/redirects.py:130]:
```suggestion
    ("ug_zigbee_commissioning","protocols/zigbee/commissioning"),
```

File: doc/nrf/ecosystem_integration.rst (hunk 3 above)
Comment (peknis) [reply doc/nrf/ecosystem_integration.rst:10]: Not fixed yet.

File: doc/nrf/ecosystem_integration.rst (hunk 4 above)
Comment (peknis) [reply doc/nrf/ecosystem_integration.rst:13]: Not fixed yet.

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by mia-ko — APPROVED** [review]

**Review by erikrobstad — APPROVED** [review]

File: doc/nrf/ecosystem_integration.rst (hunk 3 above)
Comment (b-gent) [reply doc/nrf/ecosystem_integration.rst:10]: fixed

File: doc/nrf/ecosystem_integration.rst (hunk 4 above)
Comment (b-gent) [reply doc/nrf/ecosystem_integration.rst:13]: fixed

File: doc/nrf/ecosystem_integration.rst (hunk 5 above)
Comment (b-gent) [reply doc/nrf/ecosystem_integration.rst:6]: fixed

**Comment by b-gent** [comment]
> Under protocols\thread\overview\images, ot-arch_2x.svg (the PNG version is used), thread_platform_design_nRF53.odg and thread_platform_design_nRF53_multi.odg are not used anywhere. May be we can remove these.

removed

**Comment by b-gent** [comment]
> doc/nrf/images/ncs-west-repos.png → doc/nrf/developing/images/ncs-west-repos.png. I don't think this PNG image is being used. There is a svg equivalent that is used in the docs.

removed

**Comment by b-gent** [comment]
> For [doc/nrf/app_power_opt.rst → doc/nrf/app_dev/optimizing/power_opt.rst](https://github.com/nrfconnect/sdk-nrf/pull/9561/files#diff-477f63e7ee0615ac9d66ca84b7213eb2753acaa1825aaf3c888e54bc6227defd) , do we need the _opt suffix for the new file name?

suffix removed

**Review by peknis — APPROVED** [review]

**Comment by b-gent** [comment]
> [ /images/matter_components_DM_doorlock.svg → .../images/matter_components_DM_doorlock.svg ](https://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-9561/nrf/protocols/matter/overview/data_model.html#data-model-example-door-lock) is missing the caption.

corrected in data_model.rst

**Review by wiba-nordic — APPROVED** [review]
Looked through the Thread section an didn't notice anything wrong. Nice job!

File: doc/_utils/redirects.py (hunk 7 above)
Comment (b-gent) [reply doc/_utils/redirects.py:130]: corrected

File: doc/_utils/redirects.py (hunk 6 above)
Comment (b-gent) [reply doc/_utils/redirects.py:52]: corrected

File: doc/_utils/redirects.py [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+    ("ug_bootloader_config", "app_dev/bootloaders_and_dfu/bootloader_config"),
+    ("ug_bootloader_external_flash", "app_dev/bootloaders_and_dfu/bootloader_external_flash"),
+    ("ug_bootloader_testing", "app_dev/bootloaders_and_dfu/bootloader_testing"),
+    ("ug_fw_update", "app_dev/bootloaders_and_dfu/fw_update"),
+    ("app_bootloaders", "app_dev/bootloaders_and_dfu/index"),
+    ("ug_logging", "app_dev/logging/index"),
+    ("ug_multi_image", "app_dev/multi_image/index"),
+    ("app_opt", "app_dev/optimizing/index"),
+    ("app_memory", "app_dev/optimizing/memory"),
+    ("app_power_opt", "app_dev/optimizing/power_opt"),
```
Comment (umapraseeda) [inline doc/_utils/redirects.py:32]: since we renamed the file 
```suggestion
    ("app_power_opt", "app_dev/optimizing/power"),
```

Comment (b-gent) [reply doc/_utils/redirects.py]: right, thanks :)

**Comment by umapraseeda** [comment]
Also, there is one file ncs-west-repos.xml in the images folder. Not sure if that is required.

**Review by umapraseeda — APPROVED** [review]
LGTM! Thanks!

**Comment by b-gent** [comment]
> Also, there is one file ncs-west-repos.xml in the images folder. Not sure if that is required.

not sure either, removed

**Review by richabp — APPROVED** [review]

**Review by annwoj — APPROVED** [review]

**Pull request merged**